pages/lua_manual/api_reference.md

```diff
@@ -0,0 +1,1849 @@
+
+
+
+
+### Property Descriptions
+
+
+
+### Method Descriptions
+
+* `nil`{: .type} DataSourcenew()
+
+> 
+
+
+
+## ElementFormControlDataSelect
+
+Inherits: `ElementFormControlSelect`{: .type }
+
+ElementFormControlDataSelect derives from ElementFormControlSelect. It has the following additional method:
+
+### Properties
+
+
+
+
+### Methods
+
+| Return value | Name |
+| ------------ | ---- |
+| `nil`{: .type } | SetDataSource(`string`{: .type } data_source_name) |
+
+
+### Metafunctions
+
+
+
+
+### Property Descriptions
+
+
+
+### Method Descriptions
+
+* `nil`{: .type} SetDataSource(`string`{: .type } data_source_name)
+
+> Sets the name and table of the new data source to be used by the select box.
+
+
+
+## ElementForm
+
+Inherits: `Element`{: .type }
+
+ElementForm derives from Element. The form element has the following method:
+
+### Properties
+
+
+
+
+### Methods
+
+| Return value | Name |
+| ------------ | ---- |
+| `nil`{: .type } | Submit(`string`{: .type } submit_value) |
+
+
+### Metafunctions
+
+
+
+
+### Property Descriptions
+
+
+
+### Method Descriptions
+
+* `nil`{: .type} Submit(`string`{: .type } submit_value)
+
+> Submits the form with a submit value of submit_value.
+
+
+
+## ElementText
+
+Inherits: `Element`{: .type }
+
+
+
+### Properties
+
+| Types | Name |
+| ------------ | ---- |
+| `string`{: .type } | text |
+
+
+### Methods
+
+
+
+
+### Metafunctions
+
+
+
+
+### Property Descriptions
+
+*  `string`{: .type } text
+
+> 
+
+
+
+### Method Descriptions
+
+
+
+## Element
+
+Inherits: `nil`{: .type }
+
+The Element class has no constructor; it must be instantiated through a [Document](#document) object instead. It has the following methods and properties:
+
+### Properties
+
+| Types | Name |
+| ------------ | ---- |
+| `nil`{: .type } | next_sibling |
+| `Document`{: .type } | owner_document |
+| `string`{: .type } | class_name |
+| `Element`{: .type } | offset_parent |
+| `number`{: .type } | scroll_height |
+| `number`{: .type } | scroll_left |
+| `number`{: .type } | offset_width |
+| `string`{: .type } | id |
+| `nil`{: .type } | first_child |
+| `nil`{: .type } | last_child |
+| `number`{: .type } | client_left |
+| `ElementChildNodesProxy`{: .type } | child_nodes |
+| `nil`{: .type } | previous_sibling |
+| `number`{: .type } | client_height |
+| `number`{: .type } | scroll_top |
+| `number`{: .type } | offset_top |
+| `ElementStyleProxy`{: .type } | style |
+| `number`{: .type } | scroll_width |
+| `number`{: .type } | client_top |
+| `number`{: .type } | offset_left |
+| `number`{: .type } | offset_height |
+| `string`{: .type } | tag_name |
+| `string`{: .type } | inner_rml |
+| `number`{: .type } | client_width |
+| `ElementAttributesProxy`{: .type } | attributes |
+| `nil`{: .type } | parent_node |
+
+
+### Methods
+
+| Return value | Name |
+| ------------ | ---- |
+| `nil`{: .type } | AddEventListener(`BOOL`{: .type } event, `string`{: .type } listener[, `lua_type`{: .type } in_capture_phase]) |
```

Comment:
```suggestion
| `nil`{: .type } | AddEventListener(`boolean`{: .type } event, `string`{: .type } listener[, `lua_type`{: .type } in_capture_phase]) |
```
I think this is the proper Lua terminology?

Reply:
Sure, done